### PR TITLE
Add doc to guide how to enable canary deployment for k8s app with istio

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: "Canary deployment for Kubernetes app with Istio"
+linkTitle: "Canary k8s app with Istio"
+weight: 1
+description: >
+  How to enable canary deployment for Kubernetes application with Istio.
+---
+
+"Canary release is a technique to reduce the risk of introducing a new software version in production by slowly rolling out the change to a small subset of users before rolling it out to the entire infrastructure and making it available to everybody."
```
Comment (khanhtc1202) [inline docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: nit, using quotes for a better look 👀 
```
> Canary release is a technique to reduce the risk of introducing a new software version in production by slowly rolling out the change to a small subset of users before rolling it out to the entire infrastructure and making it available to everybody.
```

**Comment by khanhtc1202** [comment]
/lgtm
just a nit comment 👍

File: docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md (hunk 1 above)
Comment (nghialv) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Sure. Which style do you prefer?

A.
Screen Shot 2021-02-02 at 11 39 55

B.
Screen Shot 2021-02-02 at 11 41 00

File: docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+![](/images/example-canary-kubernetes-istio.png)
+<p style="text-align: center;">
+Deployment Details Page
+</p>
+
+- Now you have an automated canary deployment for your application. 🎉
+
+## Understanding what happened
+
+In this example, you configured the deployment configuration file (`.pipe.yaml`) to migrate traffic from an old to new version of application using Istio's weighted routing feature.
```
Comment (nakabonne) [inline docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:101]: nit 
```suggestion
In this example, you configured the deployment configuration file (`.pipe.yaml`) to migrate traffic from an old to a new version of the application using Istio's weighted routing feature.
```

File: docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+![](/images/example-canary-kubernetes-istio-stage-1.png)
+
+- Stage 2: `K8S_TRAFFIC_ROUTING` ensures that 10% of traffic should be routed to canary variant and 90% to primary variant. Because the `trafficRouting` is configured to use Istio, PipeCD will find Istio's VirtualService resource of this application to control the traffic percentage.
+
+![](/images/example-canary-kubernetes-istio-stage-2.png)
+
+- Stage 3: `WAIT_APPROVAL` waits for a manual approval from someone in your team.
+
+- Stage 4: `K8S_PRIMARY_ROLLOUT` ensures that all resources of primary variant will be updated to new version.
```
Comment (nakabonne) [inline docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:114]:
```suggestion
- Stage 4: `K8S_PRIMARY_ROLLOUT` ensures that all resources of primary variant will be updated to a new version.
```

File: docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md (hunk 1 above)
Comment (nakabonne) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: I don't know how to be shown on the hugo server exactly but sounds good.

Comment (nakabonne) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Again, browser refresh stuff.

Can't be nested quotes on Hugo right?

Comment (khanhtc1202) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: @nakabonne refresh your chrome tab and you will see @nghialv suggestion example 😂 
How about this https://stackoverflow.com/a/2002150/5938316

```
> Quote here.
>
> -- <cite>Benjamin Franklin</cite>
```

Comment (nghialv) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Nice. How about this? Looks good.
Screen Shot 2021-02-02 at 11 57 36

Comment (khanhtc1202) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Cool 👍 But I think to add `from` as `from martinfowler.com/canaryrelease` maybe better 😂

Comment (nghialv) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Screen Shot 2021-02-02 at 12 02 06

Comment (nghialv) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Maybe this is better. Don't need "from" because of the same line.

Comment (nghialv) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: Or in the new line with "From". lol

Comment (khanhtc1202) [reply docs/content/en/docs/user-guide/examples/k8s-app-canary-with-istio.md:9]: I will go for this 💯 
https://github.com/pipe-cd/pipe/pull/1528#discussion_r568290360

**Comment by nghialv** [comment]
Thanks. I fixed all of them.

**Comment by khanhtc1202** [comment]
Nice 👍 
/lgtm

**Comment by nakabonne** [comment]
Thank you. Looks pretty cool.
/approve

**Pull request merged**